tests/Checker/Checks_Tests.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Tests for the Checks class.
+ *
+ * @package plugin-check
+ */
+
+use Exception;
+use WordPress\Plugin_Check\Checker\Checks;
+
+class Checks_Tests extends WP_UnitTestCase {
+
+	protected $checks;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->checks = new Checks( 'test-plugin/test-plugin.php' );
+	}
+
+	public function test_prepare_returns_callable() {
+		$cleanup = $this->checks->prepare();
+
+		$this->assertIsCallable( $cleanup );
+	}
+
+	public function test_get_checks_returns_array_of_expected_checks() {
+		$expected = array(
+			'example_check_class',
+		);
+
+		add_filter(
+			'wp_plugin_check_checks',
+			function( $checks ) use ( $expected ) {
+				return $expected;
+			}
+		);
+
+		$checks = $this->checks->get_checks();
+
+		$this->assertIsArray( $checks );
+		$this->assertSame( $expected, $checks );
+	}
+
+	public function test_run_all_checks_throws_exception_if_not_prepared() {
+		$this->expectException( Exception::class );
+
+		$this->checks->run_all_checks();
+	}
+
+	public function test_run_single_check_throws_exception_if_not_prepared() {
+		$this->expectException( Exception::class );
+
+		$this->checks->run_single_check( 'check' );
+	}
```

Comment:
Can you also add tests for these two methods when the environment _is_ prepared? Right now we only test the error case, but it's arguably even more important to test the success case.

If you had those tests, it would also have shown the PHP error in the `Checks` class above.